src/utils.php

```diff
@@ -645,3 +645,87 @@ function ensure_dir( $dir ) {
 
 	return realpath( $dir );
 }
+
+/**
+ * Recursively check permissions in a directory and confirm the UID/GID is properly set.
+ *
+ * @param string $dir The directory path to check.
+ *
+ * @return string|bool Error message or true if all checks pass.
+ */
+function check_permissions_in_directory_recursively( string $dir ) {
+	// Open the directory.
+	if ( $handle = opendir( $dir ) ) {
```

Comment:
Refactor for early bail here: if you cannot read the dir, you should `exit` with an error, not silently fail.

Reply:
Also: early bail if the machine is not Linux.